Assets/Scripts/Game/MagicAndEffects/Effects/Special/SanguineRoseEffect.cs

```diff
@@ -0,0 +1,76 @@
+// Project:         Daggerfall Tools For Unity
+// Copyright:       Copyright (C) 2009-2019 Daggerfall Workshop
+// Web Site:        http://www.dfworkshop.net
+// License:         MIT License (http://www.opensource.org/licenses/mit-license.php)
+// Source Code:     https://github.com/Interkarma/daggerfall-unity
+// Original Author: Numidium
+// Contributors:    
+// 
+// Notes:
+//
+using UnityEngine;
+using DaggerfallWorkshop.Game.Entity;
+using DaggerfallWorkshop.Utility;
+using DaggerfallWorkshop.Game.UserInterfaceWindows;
+using System.Collections.Generic;
+using DaggerfallWorkshop.Game.Items;
+using DaggerfallConnect.FallExe;
+using System;
+using DaggerfallConnect;
+using DaggerfallWorkshop.Game.Utility;
+
+namespace DaggerfallWorkshop.Game.MagicAndEffects.MagicEffects
+{
+    /// <summary>
+    /// Used by the Sanguine Rose. Spawns a Daedroth ally.
+    /// </summary>
+    public class SanguineRoseEffect : BaseEntityEffect
+    {
+        public static readonly string EffectKey = "SanguineRoseEffect";
+
+        public override void MagicRound()
+        {
+            base.MagicRound();
+            if (GameManager.Instance.AreEnemiesNearby())
```

Comment:
I'd prefer if this was switched to nearby objects system using

    List<PlayerGPS.NearbyObject> nearbyEnemies = GameManager.Instance.PlayerGPS.GetNearbyObjects(PlayerGPS.NearbyObjectFlags.Enemy);

GetNearbyObjects() is just doing a Linq query rather than a scene search. It's ideal for magic effects that need to scan scene frequently, as all those recursive GameObject searches add up once player is stacking many effects that do so. The nearby objects system was built to help with this.

Check out DetectEnemy and DispelUndead for some examples of use. I also use it in PassiveItemSpecialsEffect for vampirically draining nearby enemies. It's better if everything queries a centralised list rather than performing their own recursive scene search. I'm likely to move over other systems (except resting spawn detection) to nearby objects eventually.

Cheers! :)